atom/browser/browser_win.cc

```diff
@@ -125,15 +125,47 @@ bool Browser::SetUserTasks(const std::vector<UserTask>& tasks) {
   return SUCCEEDED(destinations->CommitList());
 }
 
-bool Browser::RemoveAsDefaultProtocolClient(const std::string& protocol) {
-  if (protocol.empty())
-    return false;
+std::wstring protocolLaunchPath(std::string protocol, mate::Arguments* args) {
+  // Read in optional exe path arg
+  std::wstring exePath;
+  std::string rawExePath;
 
   base::FilePath path;
-  if (!PathService::Get(base::FILE_EXE, &path)) {
-    LOG(ERROR) << "Error getting app exe path";
-    return false;
+
+  if (!args->GetNext(&rawExePath)) {
+    if (!PathService::Get(base::FILE_EXE, &path)) {
+      LOG(ERROR) << "Error getting app exe path";
+      return L"";
+    }
+    // Executable Path
+    exePath = path.value();
+  } else {
+    exePath.assign(rawExePath.begin(), rawExePath.end());
+  }
+
+  // Read in optional args arg
+  std::vector<std::string> launchArgs;
```

Comment:
Currently `native_mate` doesn't have a converter for `std::wstring`.  Is there a way I don't know about to get a `std::wstring` from the argument list?